src/librustc_resolve/lib.rs

```diff
@@ -2296,13 +2322,32 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                                           ty.span,
                                           ResolutionError::SelfUsedOutsideImplOrTrait);
                         } else {
+                            let segment = path.segments.last();
+                            let segment = segment.expect("missing name in path");
+                            let name = segment.identifier.name;
+
+                            let candidates =
+                                self.lookup_candidates(
+                                    name,
+                                    Namespace::TypeNS,
+                                    |def: Def| match def {
+                                        Def::Trait(_)   => true,
+                                        Def::Enum(_)    => true,
+                                        Def::Struct(_)  => true,
+                                        Def::TyAlias(_) => true,
+                                        _               => false,
```

Comment:
It's most idiomatic to not vertically align match arrows since it makes for cleaner diffs. Right now, one would have to change the vertical alignment of all arrows after removing the `Def::TyAlias(_)` arm, for example.